disttask/loaddata/dispatcher.go

```diff
@@ -66,6 +156,19 @@ func (h *flowHandle) OnTicker(ctx context.Context, task *proto.Task) {
 	h.lastSwitchTime.Store(time.Now())
 }
 
+func (h *flowHandle) registerTask(ctx context.Context, task *proto.Task) {
+	val, _ := h.taskInfoMap.LoadOrStore(task.ID, &taskInfo{taskID: task.ID})
+	info := val.(*taskInfo)
+	info.register(ctx)
+}
+
+func (h *flowHandle) unregisterTask(ctx context.Context, task *proto.Task) {
```

Comment:
will the framework make sure no registerTask is called after unregisterTask is returned? And will they be concurrently called?

Reply:
if the task finished or on error, it will call `OnTicker`, see `detectTask`

same task no, different task yes.